### DISCUSSION TITLE
how to deal with .its that have the wrong time

### DISCUSSION THREAD

**Discussion by alecristia** [LAAC-LSCP/ChildProject#305] [Ideas]
Inside the .its, times are based on a universal time zone, and a parameter in the header specifies the time zone.

However, it happens sometimes that a file is processed with the wrong time zone -- either completely off (eg a PNG file processed in France will be many hours off) or one hour off (eg a file from the summer is processed in the winter in a place that applies daylight savings). 

When we notice this, we could:
a) "fix" the .its - but this means messing with it, and ideally we never do that
b) have a procedure for dealing with this similar to anonymization, where the original .its is kept and a new .its, with the correct time zone, is generated
c) have a procedure for correcting this only in the imported .csv

(b) sounds like the cleanest option, right?

**Comment by lucasgautheron** [comment]
Yes, I think b is best.

I would recommend the following structure:

```bash
annotations/its/original/raw/ # original its
annotations/its/original/raw/timezone_fixes.csv
annotations/its/raw/ # fixed its
annotations/its/converted/ # only the fixed version is imported
code/its_fix.py
```

Running `python code/its_fix.py` would fix the original its using corrections documented into `annotations/its/original/raw/timezone_fixes.csv` and then output the result in `annotations/its/raw/`.

Then the whole process is documented by the code and the CSV that contains the corrections.

What do you think?

**Reply by alecristia** [reply]
I like it! We'll get to try it soon, since i'm now importing all of the lyon files, so some of them will be 1 hour off. I'm adding this to the list of things to do with the new data manager!

**Comment by alecristia** [comment]
Actually, I found how to do it via the LENA interface, so I propose that for now we do NOT work on this feature, but instead use the LENA interface to re-generate the .its so that they have the right time zone.

**Comment by lucasgautheron** [comment]
Agreed!

Le mar. 19 oct. 2021 à 11:24, Alex Cristia ***@***.***> a
écrit :

> Actually, I found how to do it via the LENA interface, so I propose that
> for now we do NOT work on this feature, but instead use the LENA interface
> to re-generate the .its so that they have the right time zone.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/LAAC-LSCP/ChildProject/discussions/305#discussioncomment-1501269>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABI4QHXQP3B7IHC7VX6MYN3UHU2MJANCNFSM5F64YVJA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Lucas Gautheron
06 79 23 86 47

Laboratoire de Sciences Cognitives et Psycholinguistique
Bureau 414 - 29 rue d'Ulm
75005 Paris
